.github/workflows/downstream-tests-on-comment.yml

```diff
@@ -0,0 +1,183 @@
+name: downstream-tests-on-comment
+
+on:
+  issue_comment:
+    types:
+    - created
+
+jobs:
+  get-pr-sha:
+    if: ${{ github.event.issue.pull_request && github.event.comment.body == '/test-downstream'}}
+    runs-on: ubuntu-latest
+    
+    outputs:
+      sha: ${{ steps.sha.outputs.result }}
+    
+    steps:
+    - name: Get PR SHA
+      id: sha
+      uses: actions/github-script@v4
+      with:
+        result-encoding: string
+        script: |
+          const { owner, repo, number } = context.issue;
+          const pr = await github.pulls.get({
+            owner,
+            repo,
+            pull_number: number,
+          });
+          return pr.data.head.sha
+
+  unit-tests:
+    runs-on: ubuntu-latest
+    needs: get-pr-sha
+    continue-on-error: true
+    
+    name: ${{ matrix.os }} (${{ matrix.r }} (${{ matrix.module }})
+
+    strategy:
+      fail-fast: false
+      matrix:
+        r: [4.1.0]
+        os: [ubuntu-latest, macOS-latest, windows-latest]
+        # only common modules
+        module: [jasp-stats/jaspDescriptives, jasp-stats/jaspTTests, jasp-stats/jaspAnova, jasp-stats/jaspRegression, jasp-stats/jaspFrequencies, jasp-stats/jaspFactor]
+        # for debugging
+#         os: [ubuntu-latest]
+#         module: [jasp-stats/jaspDescriptives, jasp-stats/jaspTTests]
+
+#    # outputs is generated with the following R code. It does not matter if some outputs are unused.
+#     oses      <- c("ubuntu-latest", "macOS-latest", "windows-latest")
+#     modules   <- c("jasp-stats/jaspDescriptives", "jasp-stats/jaspTTests", "jasp-stats/jaspAnova", "jasp-stats/jaspRegression", "jasp-stats/jaspFrequencies", "jasp-stats/jaspFactor")
+
+#     base <- "      %1$s-%2$s: ${{ steps.update-output.outputs.%1$s-%2$s }}"
+#     for (os in oses) for (module in modules)
+#       cat(sprintf(base, os, basename(module)), "\n")
+    outputs:
```

Comment:
so you ggenerated this then?

Reply:
Yes, the R code is directly above it. There *might* be other ways to do this automatically, but I just couldn't make it work. That probably involves having the first job generate some JSON output that is then interpreted by the the next job. At that point, things get quite complex.

The main advantage of having this being generated automatically, is that it would become easier to do something like
```
/test-downstream vandenman/jaspDescriptives
```
which would allow us to:
- test one or a few modules
- specify the remote for a particular module.

This should be possible but I just don't think the benefits outweigh the time it takes for me to write the action. I'd have to use the first job to parse the input comment and generate the os/ module matrix dynamically as JSON and then pass that to the next. Dynamic stuff also introduces the danger of script injection, which we'd have to think about. For example, people could install a module that just emails a repo's GitHub secrets (PAT, etc.) to themselves.

Reply:
This solution is fine!

But wouldn't the onLoad in a jasp pkg always allow for script execution and querying environment etc?
And Im sure you can send emails from R

Reply:
>But wouldn't the onLoad in a jasp pkg always allow for script execution and querying environment etc?
>And Im sure you can send emails from R

Yes, but right now, the malicious code needs to be a part of jasp-stats/module. If you can specify a custom fork for a module, then you could comment something like `/test-downstream haxor/jaspDescriptives` where the fork `haxor/jaspDescriptives` does all kinds of bad things, like spilling secrets.